datacube/cfg/api.py

```diff
@@ -0,0 +1,301 @@
+# This file is part of the Open Data Cube, see https://opendatacube.org for more information
+#
+# Copyright (c) 2015-2023 ODC Contributors
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+Datacube configuration
+"""
+import os
+import warnings
+
+from os import PathLike
+from typing import Any, TypeAlias, Union
+
+from .cfg import find_config, parse_text
+from .exceptions import ConfigException
+from .opt import ODCOptionHandler, AliasOptionHandler, IndexDriverOptionHandler
+from .utils import ConfigDict, check_valid_env_name
+
+
+# TypeAliases for more concise type hints
+# (Unions required as typehint | operator doesn't work with string forward-references.
+GeneralisedPath: TypeAlias = str | PathLike | list[str | PathLike] | None
```

Comment:
nothing in the name of those suggests being optional, yet they are, and there are no "non-optional" versions of those, that one might need at some point. Maybe moving `| None` to use site would be more informative and more "future proof".

Reply:
Hmmm..  That's part of what I was trying to capture with "generalised", but yeah, fair point.

Reply:
I'm using `Maybe` prefix for those, like `MaybeNodata`, where `Nodata` is another alias without `| None` option.